File: tests/07_phases_ompss.dg/fortran/success_task_10.f90
! <testinfo>
! test_generator=config/mercurium-ompss
! </testinfo>
PROGRAM MAIN
  IMPLICIT NONE

  INTEGER, ALLOCATABLE :: X(:, :), Y(:, :)

  ALLOCATE(X(10, 10), Y(10, 10))

  !$OMP TASK OUT(X)
  X = 1
  !$OMP END TASK

  !$OMP TASK IN(X) OUT(Y)
  Y = X + 3
  !$OMP END TASK

  !$OMP TASKWAIT

  IF (ANY(Y(:, :) /= 4)) STOP 1
END PROGRAM MAIN

File: tests/07_phases_omp.dg/common/c/success_task_10.c
/*
<testinfo>
test_generator=config/mercurium-omp
</testinfo>
*/

void f(int x, int y)
{
#pragma omp task
    {
        struct A
        {
            int x;
            int y;
        };

        struct A a;
        a.x = x;
        a.y = y;
    }
}

int main(int argc, char* argv[])
{
    f(3, 4);

    return 0;
}
